core/storage/collection/gae_models_test.py

```diff
@@ -507,15 +538,17 @@ def test_get_all_non_private_commits(self):
         self.assertEqual('collection-b-0', commits[0][0].id)
 
     def test_get_all_non_private_commits_with_invalid_max_age(self):
-        with self.assertRaisesRegexp(
+        # type: () -> None
+        with self.assertRaisesRegexp( # type: ignore[no-untyped-call]
             Exception,
             'max_age must be a datetime.timedelta instance or None.'):
             (
                 collection_models.CollectionCommitLogEntryModel
                 .get_all_non_private_commits(
-                    2, None, max_age='invalid_max_age'))
+                    2, None, max_age='invalid_max_age')) # type: ignore[arg-type]
```

Comment:
Yes, I don't see why we need to test this if the type checks would catch cases when a string is passed to that method erroneously. (Will they?)

Reply:
Yes mypy would definitely catch type mis-matches. 

There's one catch here: Our codebase it still partially annotated. Any type annotated method in the codebase which tries to call this method (`get_all_non_private_commits` here) with invalid type will throw mypy errors - but until a calling function has been type annotated, it can still possibly pass wrong argument types to `get_all_non_private_commits` (it's a possibility, can't really comment on it's probability).

My reasoning to tackle this is - if after removing these tests we don't see any coverage error then we shouldn't be  worrying about removing tests, because it means the same lines of code which was meant to handle error is being executed in some other manner - meaning it was able to handle error via a different situation. I don't see any additional catch there. On the other hand if we do see coverage error, then I cannot precisely predict what case that might be, and it can be discussed.

What are your views on this? @seanlip @vojtechjelinek @hardikkat24 If you don't see any other problems here, I think we should start removing such tests (which use wrong argument types). Thanks!

Reply:
I'm fine with removing -- just do a general check, before doing so, to verify that wrong arg types can't be passed, based on the current code in the codebase. Thanks!

Reply:
@seanlip  @sajalasati  I think removing tests which check for wrong type of arguments is a nice idea. But there is one thing that we miss here.
Example: Let's say File A is typed and has a function `func(x: int)`. Then we have a FIle B which is untyped and imports file A and used `A.func`.
Now we remove the test from `A_test` which checks the behaviour when wrong type value is passed. Still we can use `A.func('string')` in file B as it is not type checked.
So my opinion on this case would be to remove the tests which check for bad type values when our full codebase is type annotated.

Reply:
I'm OK either way. If we are removing them later, though, there should be an issue with full context, and TODOs pointing to that issue.

It's also fine with me to remove them now if you can check that the current codebase doesn't have any possibility of calling them incorrectly, and it is unlikely that such cases will be added in the future.

Reply:
Added a TODO for this.